### PR TITLE
fix

### PR DESCRIPTION
**Pull request by efrathartman**

### CODE REVIEW

**Review by Shoshana-Yudelevitz — COMMENTED** [review]
efrat branch

File: client-side/src/App.jsx [hunk 1]
```diff
@@ -14,6 +15,7 @@ function App() {
       <SnackbarProvider maxSnack={3}>
       <Provider store={store}>
         <RouterProvider router={router} />
+        <SignUp/>
```
Comment (EA171) [inline client-side/src/App.jsx:18]: why you should added this to the App file?

File: client-side/src/axios/middleware.js [hunk 2]
```diff
@@ -9,10 +9,17 @@ export async function handleGet(path) {
 };
 
 export async function handlePost(path, data) {
-    const response = await axios.create({ baseURL: url }).post(path, data);
+    console.log(path,data,"ooo");
+    const response = await axios.create({baseURL:url}).post(path,data,
```
Comment (EA171) [inline client-side/src/axios/middleware.js:13]: please remove console.logs and format the file

File: client-side/src/components/signUp/PasswordStrength.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Box, LinearProgress, Typography } from '@mui/material';
+const PasswordStrengthMeter = ({ password }) => {
+  const PASSWORD_STRENGTH = {
```
Comment (EA171) [inline client-side/src/components/signUp/PasswordStrength.jsx:4]: this enum should be in constants file.
and the messages should be more detailed, so the user will be able to understand his mistake

File: client-side/src/components/signUp/PasswordStrength.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  };
+  const calculatePasswordStrength = (password) => {
+    if (!password) return PASSWORD_STRENGTH.EMPTY;
+    if (password.length < 4) return PASSWORD_STRENGTH.WEAK;
+    if (/[A-Za-z]/.test(password) && /[0-9]/.test(password) && password.length > 4) return PASSWORD_STRENGTH.MEDIUM;
+    return PASSWORD_STRENGTH.WEAK;
+  };
+  const strength = calculatePasswordStrength(password);
+  const { color, message } = strength; // הפקת צבע והודעה מקבועים
```
Comment (EA171) [inline client-side/src/components/signUp/PasswordStrength.jsx:28]: please remove comments

File: client-side/src/components/signUp/PasswordStrength.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <Box className="centered-container" sx={{width:"50%",marginLeft:"2%"}}>
+      <LinearProgress
+        variant="determinate"
+        value={(strength.key / 3) * 100}
+        sx={{
+          height: 10,
```
Comment (EA171) [inline client-side/src/components/signUp/PasswordStrength.jsx:35]: please move styles to scss file and colors should come from variables file

File: client-side/src/components/signUp/signUp.jsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+});
+function SignUp() {
+  const [password, setPassword] = useState('');
+  const dispatch = useDispatch();
+  const formik = useFormik({
+    initialValues: { name: '', email: '', password: '' },
+    validationSchema: SignUpSchema,
+    onSubmit: (values) => {
+      userSignUp({
+        name: values.name,
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:40]: Why are you assigning the values object to a new object? they are exactly the same

File: client-side/src/components/signUp/signUp.jsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        email: values.email,
+        password: values.password,
+      });
+    },
+  });
+  const userSignUp = async (user) => {
+    try {
+  const response = await createUser(user); 
+      localStorage.setItem("nameUser", user.name);
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:49]: why you are using localStorage? use redux

File: client-side/src/components/signUp/signUp.jsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+      dispatch(addUser(response.user));
+
+      window.location.reload();
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:52]: what is this reload?

File: client-side/src/components/signUp/signUp.jsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+      
+    } catch (error) {
+      console.error("error");
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:55]: please add more details about the failed

File: client-side/src/components/signUp/signUp.jsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  };
+  const [robotPass, setRobotPass] = useState(null)
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:58]: move all states declaration to be in the start of the component

File: client-side/src/components/signUp/signUp.jsx [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  return (
+    <div className="signup-container">
+      <h2 >signUp </h2>
+      <form onSubmit={formik.handleSubmit}>
+        <div className="form-group">
+            <GenericInput
+            label="user name"
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:65]: labels should be in constants file - do it for all

File: client-side/src/components/signUp/signUp.jsx [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+            type="text"
+            name="name"
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+            value={formik.values.name}
+            width="100%"
+            size="medium"
+            style={{ height: '50px', marginBottom: '16px' }}
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:73]: styles should be in scss file - do it for all

File: client-side/src/components/signUp/signUp.jsx [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+          />
+          {formik.touched.password && formik.errors.password ? (
+            <div className="error">{formik.errors.password}</div>
+          ) : null}
+          <PasswordStrengthMeter  password={password} />
+        </div>
+      <ReCAPTCHA
+          sitekey='6Ld5uBoqAAAAAKwPXqo5eanm9ZFSuOoBBSdl00pE'
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:116]: move this key to env

File: client-side/src/components/signUp/signUp.scss [hunk 14]
```diff
@@ -0,0 +1,60 @@
+
+/* src/SignUp.css */
+.signup-container {
+    max-width: 25%;
+    margin: 0 auto;
+    padding: 1%;
+    border: 1px solid #ccc;
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.scss:7]: colors should be in variables file

File: client-side/src/components/signUp/signUp.scss
Comment (EA171) [inline client-side/src/components/signUp/signUp.scss:1]: this file is looks like css not scss

File: client-side/src/services/userService.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
+import { handlePost } from '../axios/middleware';
+
+export const createUser = async (userData) => {
+    try {
+        console.log(userData,"userdata");
```
Comment (EA171) [inline client-side/src/services/userService.js:5]: remove console.logs

File: client-side/src/stories/GenericInput/genericInput.scss [hunk 16]
```diff
@@ -7,7 +7,9 @@
     transition: border-color 0.3s ease, box-shadow 0.3s ease;
     border-color: $color-border;
     border-width: 1px;
-    width: 100%;
+    width: 50%;
+    margin-right: 200px;
```
Comment (EA171) [inline client-side/src/stories/GenericInput/genericInput.scss:11]: don't use px, only %

File: client-side/src/stories/header/header.jsx [hunk 17]
```diff
@@ -8,6 +8,8 @@ import MessageIcon from './Icon'
 import './header.scss';
 import { selectAuth } from '../../redux/auth/auth.selector';
 function ResponsiveAppBar() {
+  const [currentname, setCurrentname] = useState(localStorage.getItem('nameUser') || null);
```
Comment (EA171) [inline client-side/src/stories/header/header.jsx:11]: why localStorage?

File: client-side/src/stories/header/header.jsx [hunk 18]
```diff
@@ -86,10 +88,14 @@ function ResponsiveAppBar() {
...
-              <Avatar>{getAvatarLetter()}</Avatar>
-              </IconButton>
+          <Tooltip title={currentname}>
+            { !currentProfile && < IconButton onClick={handleOpenUserMenu} >
+                <Avatar alt={currentname} src="/static/images/avatar/2.jpg" />
+              </IconButton>}
+        
+            { currentProfile && < IconButton onClick={handleOpenUserMenu} >
+                <Avatar alt={currentname} src="/static/images/avatar/2.jpg" />
+              </IconButton>}
```
Comment (EA171) [inline client-side/src/stories/header/header.jsx:98]: what is this duplication?
should the avatar be const? shouldn't it come from the user profile definition?

**Review by EA171 — CHANGES_REQUESTED** [review]

File: client-side/src/components/signUp/PasswordStrength.jsx (hunk 3 above)
Comment (EA171) [reply client-side/src/components/signUp/PasswordStrength.jsx]: the messages has not enough details

File: client-side/src/components/signUp/signUp.scss (hunk 14 above)
Comment (EA171) [reply client-side/src/components/signUp/signUp.scss]: variables should be defined in separate variables.scss file

File: client-side/src/App.jsx [hunk 19]
```diff
@@ -6,10 +6,10 @@ import { router } from './router/router.jsx';
 import { store } from './redux/store.jsx';
 import { SnackbarProvider } from 'notistack';
 import './App.scss';
+import SignUp from './components/signUp/signUp.jsx';
```
Comment (EA171) [inline client-side/src/App.jsx:9]: remove unused imports

File: client-side/src/components/signUp/signUp.jsx [hunk 20]
```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { useDispatch } from 'react-redux';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import './signUp.scss';
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:8]: scss imports should be the last

File: client-side/src/components/signUp/signUp.jsx [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+    validationSchema: SignUpSchema,
+    onSubmit: (values) => {
+      userSignUp(values);
+    },
+  });
+  const userSignUp = async (user) => {
+    try {
+      await createUser(user); 
+      dispatch(addUser(user));
+      window.location.href="/home"
```
Comment (EA171) [inline client-side/src/components/signUp/signUp.jsx:45]: use useNavigate instead window.location.href="/home"

**Review by EA171 — CHANGES_REQUESTED** [review]

**Review by EA171 — APPROVED** [review]

**Review by e171271 — APPROVED** [review]

**Pull request merged**